### ISSUE TITLE
Trapper/Animal Procurer - Animal Market

### ISSUE THREAD

**Issue by chaosdwarfs** [ldtteam/minecolonies-features#216]
### Prerequisites 

- [ x] I am running the latest **alpha** version of MineColonies for my Minecraft version.
- [x ] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure that my feature does not already exist.
- [x ] I made sure that **this feature request is not a duplicate of any existing feature request.**

### Context 

- Minecraft Version: 1.16
- MineColonies Version: 496 alpha

Suggestion:
Add a workers hut that lets you have access to animals for your colony- animals that are present in your world, letting you reach animals even in different biomes.

**How it would work:**
It would work in the same vein Visitors to the tavern work, but with animals 'visiting' instead. The hut-structure would resemble either an animal pen/zoo-like structure, or something closer to an animal auction. 
The Trapper/Procurer would thus bring livestock and exotic animals to your colony which, if you got the money, you can buy at steep price to start that ranch or sheepfarm, or perhaps just to give your colonists something exciting to experience.
What it would add would be a way for the player to get to animals they might not have access too, like animals on distant islands or biomes. 

Picture example:
https://thenantwichnews.co.uk/wp-content/uploads/Welsh-cattle-drovers-visit-the-Row-in-Medieval-times-by-Les-Pickford-1.jpg

In the pen a random animal would spawn, one more animal per level. 
The random chance would be perhaps be weighted towards the usual animals (cows, sheeps, pigs, etc.).
Then every 'tick' the pen has a chance to spawn X*hutlevel number of animals. 
Chances of animal spawning could perhaps be tied to worker level? So a dashing and experienced Trapper/Procurer will be more able to capture or barter for animals out in the world. 

Advanced questions and considerations:
What would happen if this system makes hostile animal mobs spawn?

1) Perhaps the worker would have a tranquilizer or some sort of friendly potion/tool that prevents animals in the pen to attack each other or attack the worker?
2) Or if the animals could be spawned in enclosed pens? 
3) Perhaps you can choose hostile animals not to spawn?

**Who would benefit?**
This would help immensely with players who doesn't do massive amount of shuffling animals on boats and leading them miles and miles on end. It would help Colonies in caves and other inhospitable places that have scarce access to animals. 
It would give variety to the Colony and a rewarding routine. Players waking up can take a stroll past the Animal Market, see if the pigs have finally come in, or to see what new exciting wonders have been brought in (maybe a cute parrot or a devious llama!) or to see if an especially good horse has come in that you can breed on the studfarm.

        
**TL:DR**
1. An animal market Hut
2. Animal Market spawns animals 
3. The Player can buy these animals and use the animals as if the animals found them in the world

---
## Viewers

* Add a thumbs-up to the feature request to vote for it. This helps the request become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by ShaneCourtrille** [comment]
I checked with @ravenbuilder934 as to why this got a downvote and was provided with this link that I'm posting with their permission.

https://discord.com/channels/139070364159311872/240841248750043137/786969147649032272
